device/src/test_util.rs

```diff
@@ -64,14 +65,20 @@ pub fn get_abp_credentials() -> JoinMode {
 
 pub type RxTxHandler = fn(Option<Uplink>, RfConfig, &mut [u8]) -> usize;
 
+lazy_static::lazy_static! {
+    static ref SESSION: Mutex<HashMap<usize, Session>> = Mutex::new(HashMap::new());
+
+}
+
 /// Handle join request and pack a JoinAccept into RxBuffer
-pub fn handle_join_request(
+pub fn handle_join_request<const I: usize>(
```

Comment:
I think it would make sense to have this return Result() than add so many panic!'s

Reply:
@lulf It's a helper for tests

Reply:
Ah, good to know, thanks!